packages/react-ui-components/src/components/SchemaForm/ArrayFieldTemplate/ArrayFieldTemplate.tsx

```diff
@@ -65,47 +67,75 @@ type ArrayFieldItemProps = {
 
 // Used in the two templates
 const DefaultArrayItem = ({ children, disabled, hasMoveDown, hasMoveUp, hasRemove, hasToolbar, index, onDropIndexClick, onReorderClick, readonly, key }: ArrayFieldItemProps) => {
+	const resetStyle = !get(children, ['props', 'uiSchema', 'ui:fieldset'], false);
+	const itemPrefix = get(children, ['props', 'uiSchema', 'ui:itemPrefix']);
+
 	return (
-		<div key={key} className={styles['array-item-container']}>
-			{children}
-			{hasToolbar && (
-				<div className={styles['toolbar-container']}>
-					{(hasMoveUp || hasMoveDown) && (
-						<>
-							<KvActionButtonIcon
-								icon={EIconName.AlignBottom}
-								size={EComponentSize.Large}
-								type={EActionButtonType.Secondary}
-								tabIndex={-1}
-								disabled={disabled || readonly || !hasMoveUp}
-								onClickButton={onReorderClick(index, index - 1)}
-							/>
+		<fieldset key={key} className={classNames([], { [styles['reset-fieldset-style']]: resetStyle })}>
+			<div className={styles['array-item-container']}>
+				{itemPrefix && <span className={styles['item-prefix']}>{`${itemPrefix} ${index + 1}`}</span>}
+				{children}
+				{hasToolbar && (
+					<div className={styles['toolbar-container']}>
+						{(hasMoveUp || hasMoveDown) && (
+							<>
+								<KvActionButtonIcon
+									icon={EIconName.AlignBottom}
+									size={EComponentSize.Large}
+									type={EActionButtonType.Tertiary}
+									tabIndex={-1}
+									disabled={disabled || readonly || !hasMoveUp}
+									onClickButton={onReorderClick(index, index - 1)}
+								/>
+								<KvActionButtonIcon
+									icon={EIconName.AlignTop}
+									size={EComponentSize.Large}
+									type={EActionButtonType.Tertiary}
+									tabIndex={-1}
+									disabled={disabled || readonly || !hasMoveDown}
+									onClickButton={onReorderClick(index, index + 1)}
+								/>
+							</>
+						)}
+						{hasRemove && (
 							<KvActionButtonIcon
-								icon={EIconName.AlignTop}
+								icon={EIconName.Delete}
 								size={EComponentSize.Large}
-								type={EActionButtonType.Secondary}
+								type={EActionButtonType.Tertiary}
 								tabIndex={-1}
-								disabled={disabled || readonly || !hasMoveDown}
-								onClickButton={onReorderClick(index, index + 1)}
+								disabled={disabled || readonly}
+								onClickButton={onDropIndexClick(index)}
 							/>
-						</>
-					)}
-					{hasRemove && (
-						<KvActionButtonIcon
-							icon={EIconName.Delete}
-							size={EComponentSize.Large}
-							type={EActionButtonType.Danger}
-							tabIndex={-1}
-							disabled={disabled || readonly}
-							onClickButton={onDropIndexClick(index)}
-						/>
-					)}
-				</div>
-			)}
-		</div>
+						)}
+					</div>
+				)}
+			</div>
+		</fieldset>
 	);
 };
 
+const addButton = ({ canAdd, disabled, readonly, uiSchema, onAddClick }: Partial<ArrayFieldTemplateProps>) =>
```

Comment:
when we have functional components it's better to have this constant call in PascalCase, so AddButton